akka-persistence-tck/src/main/scala/akka/persistence/PluginSpec.scala

```diff
@@ -35,10 +46,87 @@ abstract class PluginSpec(val config: Config) extends TestKitBase with WordSpecL
 
   def extension: Persistence = _extension
 
+  /**
+   * Returns a new unique persistence id
+   */
+  def nextPid: String =
+    s"p-${counter.incrementAndGet()}"
+
+  /**
+   * Returns the current persistence id
+   */
   def pid: String = _pid
 
+  /**
+   * Returns the current writer uuid
+   */
   def writerUuid: String = _writerUuid
 
+  /**
+   * Returns the actorRef of the default configured journal plugin
+   */
+  def journal: ActorRef =
+    extension.journalFor(null)
+
   def subscribe[T: ClassTag](subscriber: ActorRef) =
     system.eventStream.subscribe(subscriber, implicitly[ClassTag[T]].runtimeClass)
+
+  /**
+   * Overridable hook that is called before populating the journal for the next
+   * test case. `pid` is the `persistenceId` that will be used in the test.
+   * This method may be needed to clean pre-existing events from the log.
+   */
+  def preparePersistenceId(pid: String): Unit = ()
+
+  /**
+   * Implementation may override and return false if it does not
+   * support atomic writes of several events, as emitted by `persistAll`.
+   */
+  def supportsAtomicPersistAllOfSeveralEvents: Boolean = true
+
+  def replayedMessage(snr: Long, deleted: Boolean = false, confirms: Seq[String] = Nil): ReplayedMessage =
+    ReplayedMessage(PersistentImpl(s"a-$snr", snr, pid, "", deleted, Actor.noSender, writerUuid))
+
+  def writeMessages(fromSnr: Int, toSnr: Int, pid: String, sender: ActorRef, writerUuid: String, tags: String*): Unit = {
+    def persistentRepr(sequenceNr: Long) =
+      PersistentRepr(
+        payload = if (tags.isEmpty) s"a-$sequenceNr" else Tagged(s"a-$sequenceNr", Set(tags: _*)),
+        sequenceNr = sequenceNr,
+        persistenceId = pid,
+        sender = sender,
+        writerUuid = writerUuid
+      )
+
+    val msgs: Seq[PersistentEnvelope] =
+      if (supportsAtomicPersistAllOfSeveralEvents)
+        Seq(AtomicWrite((fromSnr to toSnr).map(i ⇒ persistentRepr(i))))
+      else (fromSnr to toSnr).map(i ⇒ AtomicWrite(persistentRepr(i)))
+
+    val probe = TestProbe()
+
+    journal ! WriteMessages(msgs, probe.ref, actorInstanceId)
+
+    probe.expectMsg(WriteMessagesSuccessful)
+    fromSnr to toSnr foreach { seqNo ⇒
+      probe.expectMsgPF() {
+        case WriteMessageSuccess(PersistentImpl(payload, `seqNo`, `pid`, _, _, `sender`, `writerUuid`), _) ⇒
+          val id = s"a-$seqNo"
+          payload should matchPattern {
+            case `id`            ⇒
+            case Tagged(`id`, _) ⇒
+          }
+      }
+    }
+  }
+
+  /**
+   * Deletes messages from the journal.
+   */
+  def delete(persistenceId: String, toSequenceNr: Long = Long.MaxValue): Unit = {
+    val probe = TestProbe()
+    journal ! DeleteMessagesTo(persistenceId, toSequenceNr, probe.ref)
```

Comment:
I changed it, also I used the ` probe.expectMsgType[DeleteMessagesSuccess](waitAtMost)` call.